src/Makefile.am

```diff
@@ -1691,6 +1692,7 @@ tests_testACLMaxUserIP_SOURCES = \
 	tests/testACLMaxUserIP.cc \
 	tests/testACLMaxUserIP.h
 nodist_tests_testACLMaxUserIP_SOURCES = \
+	tests/stub_CachePeer.cc \
```

Comment:
Done in 847ea49.

Reply:
> These lists should really be sorted by their full string value.

@yadij, please note that your definition of correct `_SOURCES` entry order differs from the one currently used by `scripts/source-maintenance.sh` (or by `scripts/format-makefile-am.pl`, to be precise). The current automated order is based on the `$filename.$prefix` expression, where `$prefix` is `tests/stub_` or `tests/test`.

If you think the automated order is wrong, then please post a PR to change `scripts/format-makefile-am.pl`.

FWIW, I would support simplifying the currently overly complex `_SOURCES` sorting rules, migrating to the "full string value" algorithm you have mentioned. Such migration would allow sorting these fields in text editors. It would also make the algorithm much simpler and robust. The resulting loss of spacial proximity of related stubbed/test/full files is an acceptable sacrifice in this case IMHO.